app/Http/Controllers/ConferencesController.php

```diff
@@ -111,14 +112,28 @@ public function show($id)
             return redirect('/');
         }
 
-        $talksAtConference = $conference->myTalks()->map(function ($talkRevision) {
-            return $talkRevision->talk->id;
+        $submissions = $conference->mySubmissions();
+        $acceptances = $conference->myAcceptedTalks();
+
+        $talks = auth()->user()->talks->map(function ($talk) use ($submissions, $acceptances) {
+            /** @var Talk $currentTalk */
+            $currentTalk = $talk->current();
+            return [
+                'id' => $talk->id,
+                'title' => $currentTalk->title,
+                'url' => $currentTalk->getUrl(),
+                'submitted' => $submissions->search(function ($item) use ($talk) {
```

Comment:
This could be moved onto the model, yah?

`$talk->submittedToConference($conference)`

same for accepted:

`$talk->acceptedToConference($conference)`

then they would have the same logic you have here.

Reply:
:+1: I think that's cleaner, yes.